src/test/java/net/oauth/jsontoken/JsonTokenParserTest.java

```diff
@@ -435,21 +435,20 @@ private JsonToken getJsonTokenWithTimeRange(Instant issuedAt, Instant expiration
     return checkToken;
   }
 
+  // deserializes the token string such that tokenStrings without signatures are allowed
   private JsonToken naiveDeserialize(String tokenString) {
-    assertTrue(
-        tokenString.equals(TOKEN_STRING_2PARTS)
-          || tokenString.equals(TOKEN_STRING_EMPTY_SIG)
-    );
-
-    List<String> pieces = Splitter.on(JsonTokenUtil.DELIMITER).splitToList(TOKEN_STRING_2PARTS);
+    List<String> pieces = Splitter.on(JsonTokenUtil.DELIMITER).splitToList(tokenString);
+    if (pieces.size() < 2) {
+      fail("naiveDeserialize does not support deserializing the token string: " + tokenString);
+    }
```

Comment:
nit: Just remove the input checking altogether. It's already named "naive" and is only used in tests. I prefer to use the junit assert/fail methods for asserting things about what's actually being tested, rather than making sure the test utility functions are being used properly. You could update the javadoc saying it won't work with less than 2 parts.

Reply:
I moved the description above in the javadoc.